x/tbr/internal/keeper/keeper.go

```diff
@@ -0,0 +1,191 @@
+package keeper
+
+import (
+	"github.com/commercionetwork/commercionetwork/x/tbr/internal/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/exported"
+)
+
+type Keeper struct {
+	StoreKey sdk.StoreKey
+
+	BankKeeper         bank.Keeper
+	StakeKeeper        staking.Keeper
+	DistributionKeeper distribution.Keeper
+
+	Cdc *codec.Codec
+}
+
+func NewKeeper(storeKey sdk.StoreKey, bk bank.Keeper, sk staking.Keeper, dk distribution.Keeper, cdc *codec.Codec) Keeper {
+	return Keeper{
+		StoreKey:           storeKey,
+		BankKeeper:         bk,
+		StakeKeeper:        sk,
+		DistributionKeeper: dk,
+		Cdc:                cdc,
+	}
+}
+
+//Utility method to set Block Reward Pool
+func (k Keeper) SetBlockRewardsPool(ctx sdk.Context, updatedPool sdk.DecCoins) {
+	store := ctx.KVStore(k.StoreKey)
+	store.Set([]byte(types.BlockRewardsPoolPrefix), k.Cdc.MustMarshalBinaryBare(&updatedPool))
+}
+
+//Return the Block Rewards Pool if exists
+func (k Keeper) GetBlockRewardsPool(ctx sdk.Context) sdk.DecCoins {
+	var brPool sdk.DecCoins
+	store := ctx.KVStore(k.StoreKey)
+	brpBz := store.Get([]byte(types.BlockRewardsPoolPrefix))
+	k.Cdc.MustUnmarshalBinaryBare(brpBz, &brPool)
+	return brPool
+}
+
+//Increase the Block Rewards Pool with the specified coin amount
+func (k Keeper) IncrementBlockRewardsPool(ctx sdk.Context, funder sdk.AccAddress, amount sdk.Coins) error {
+	bk := k.BankKeeper
+	brPool := sdk.DecCoins{}
+
+	if bk.HasCoins(ctx, funder, amount) {
```

Comment:
I personally think that, if the `bk.HasCoins` returns `false`, we should return an error. Otherwise, completing without any error might signal the user that the fund has been approved even if no coins have been moved from its wallet into the pool. Does it sound good?

Reply:
Ye, your right, didn't think about that before